### PR TITLE
chore: fix compiler warnings in clientruntime

### PR DESCRIPTION
**Pull request by wooj2**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift [hunk 1]
```diff
@@ -37,7 +37,7 @@ public class CRTClientEngine: HttpClientEngine {
             logger.error("Server name was not able to be set in TLS Connection Options. TLS Negotiation will fail.")
             logger.error("Error: \(err.localizedDescription)")
         }
-        var socketOptions = SocketOptions(socketType: .stream)
+        let socketOptions = SocketOptions(socketType: .stream)
```
Comment (kneekey23) [inline Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift:40]: You can still adjust connect timeout on iOS even if this is let?

File: Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift [hunk 2]
```diff
@@ -37,7 +37,7 @@ public class CRTClientEngine: HttpClientEngine {
...
 #if os(iOS) || os(watchOS)
```
Comment (kneekey23) [inline Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift:41]: Can you add tvOS ti this line while you're in here please?

**Review by kneekey23 — APPROVED** [review]
2 questions but LGTM

File: Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift (hunk 1 above)
Comment (wooj2) [reply Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift:40]: Seems that way :)

Comment (wooj2) [reply Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift:40]: Let me triple check that we can do that on iOS

Comment (wooj2) [reply Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift:40]:
```
        let socketOptions = SocketOptions(socketType: .stream)
#if os(iOS) || os(watchOS)
        socketOptions.connectTimeoutMs = 30_000
#endif
```

Seems to compile with iOS, so i think we are good.  Here is how i tested it.
1.  Compiled the following for mac:
```
        let socketOptions = SocketOptions(socketType: .stream)
#if os(iOS) || os(watchOS)
alkdjfklasjflksjdf
        socketOptions.connectTimeoutMs = 30_000
#endif
```
Result: build successful

1.  Compiled the following for iOS:
```
        let socketOptions = SocketOptions(socketType: .stream)
#if os(iOS) || os(watchOS)
alkdjfklasjflksjdf
        socketOptions.connectTimeoutMs = 30_000
#endif
```
result: build failure

1.  Compiled the following for iOS (removed junk):
```
        let socketOptions = SocketOptions(socketType: .stream)
#if os(iOS) || os(watchOS)
        socketOptions.connectTimeoutMs = 30_000
#endif
```
result: build succeeded

Comment (kneekey23) [reply Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift:40]: nice, lets merge this then!

**Pull request merged**